### PR TITLE
feat(as-4417): dev only changes for as-4185

### PR DESCRIPTION
**Pull request by nssdesikan**
## Ticket Number
https://pins-ds.atlassian.net/browse/AS-4417

## Description of change
Develop LPA questionnaire Task-List

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/lpa-questionnaire-web-app/src/controllers/full-appeal/task-list.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+    return TASK_SCOPE.DETERMINISTIC;
+  }
+  return TASK_SCOPE.GENERIC;
+};
+
+const buildTasksList = (questionnaire, appealType, appealDecisionOutcome) => {
+  const taskList = [];
+
+  const decisionOutcome = getDecisionOutcome(appealType, appealDecisionOutcome);
```
[the remark this answers is not in the document]
Comment (nssdesikan) [reply packages/lpa-questionnaire-web-app/src/controllers/full-appeal/task-list.js:24]: Line 30 is in a loop and the decisionOutcome does not change based on the loop value

File: packages/lpa-questionnaire-web-app/src/controllers/full-appeal/task-list.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  Object.keys(SECTIONS).forEach((sectionKey) => {
+    const status = getTaskStatus(questionnaire, sectionKey);
+    const currentSection = SECTIONS[sectionKey];
+    const isSectionAvailable = currentSection.scope
```
[the remark this answers is not in the document]
Comment (nssdesikan) [reply packages/lpa-questionnaire-web-app/src/controllers/full-appeal/task-list.js:29]: Updated

File: packages/lpa-questionnaire-web-app/src/controllers/full-appeal/task-list.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        attributes: { name: sectionKey, [`${sectionKey}-status`]: status },
+        status,
+      });
+    }
+  });
+
+  return taskList;
+};
+
+const getTotalTasksCount = (sections) => {
```
[the remark this answers is not in the document]
Comment (nssdesikan) [reply packages/lpa-questionnaire-web-app/src/controllers/full-appeal/task-list.js:46]: Updated

File: packages/lpa-questionnaire-web-app/__tests__/unit/full-appeal/emptyAppealReply.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  id: null,
+  appealId: null,
+  procedureTypeReview: {},
+  issuesConstraintsDesignation: {},
+  environmentalImpactAssessment: {},
+  peoplNotification: {},
```
[the remark this answers is not in the document]
Comment (nssdesikan) [reply packages/lpa-questionnaire-web-app/__tests__/unit/full-appeal/emptyAppealReply.js:7]: Updated

File: packages/lpa-questionnaire-web-app/__tests__/unit/full-appeal/mockAppeal.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+} = require('@pins/business-rules');
+
+module.exports = {
+  aboutYouSection: { yourDetails: { name: 'Someone' } },
+  appealType: APPEAL_ID.PLANNING_SECTION_78,
+  decisionDate: null,
+  eligibility: { applicationDecision: APPLICATION_DECISION.GRANTED, enforcementNotice: null },
+  requiredDocumentsSection: { applicationNumber: '12345' },
+  appealSiteSection: {
+    siteAddress: { address1: '11 kingston Road', city: 'Bristol', postCode: 'BR12 7AU' },
```
[the remark this answers is not in the document]
Comment (nssdesikan) [reply packages/lpa-questionnaire-web-app/__tests__/unit/full-appeal/mockAppeal.js:14]: Eagle eye. Updated

File: packages/lpa-questionnaire-web-app/__tests__/unit/full-appeal/mockAppealReply.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  id: 'mock-id',
+  appealId: '89aa8504-773c-42be-bb68-029716ad9756',
+  procedureTypeReview: {},
+  issuesConstraintsDesignation: {},
+  environmentalImpactAssessment: {},
+  peoplNotification: {},
```
[the remark this answers is not in the document]
Comment (nssdesikan) [reply packages/lpa-questionnaire-web-app/__tests__/unit/full-appeal/mockAppealReply.js:7]: Updated

File: packages/lpa-questionnaire-web-app/src/services/full-appeal/task.service.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  },
+  questionnaireSubmission: {
+    displayText: 'Check your answers and submit',
+    href: '',
+    rule: () => TASK_STATUS.CANNOT_START_YET,
+  },
+};
+
+const getTaskStatus = (questionnaire, section, taskName, sections = SECTIONS) => {
+  try {
```
[the remark this answers is not in the document]
Comment (nssdesikan) [reply packages/lpa-questionnaire-web-app/src/services/full-appeal/task.service.js:65]: Updated

File: packages/lpa-questionnaire-web-app/src/controllers/full-appeal/task-list.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  if (appealType === APPEAL_ID.PLANNING_SECTION_78) {
+    if (appealDecisionOutcome === APPLICATION_DECISION.NODECISIONRECEIVED) {
+      return TASK_SCOPE.NON_DETERMINISTIC;
+    }
+    return TASK_SCOPE.DETERMINISTIC;
+  }
+  return TASK_SCOPE.GENERIC;
+};
+
+const buildTasksList = (questionnaire, appealType, appealDecisionOutcome) => {
```
[the remark this answers is not in the document]
Comment (nssdesikan) [reply packages/lpa-questionnaire-web-app/src/controllers/full-appeal/task-list.js:21]: Updated

**Review by swdpcomputing — APPROVED** [review]

**Pull request merged**